src/Sarif.Converters/BuiltInConverterFactory.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.Sarif.Converters
+{
+    // Factory class for creating one of the built-in converters.
+    internal class BuiltInConverterFactory : ConverterFactory
+    {
+        private static readonly IDictionary<string, Lazy<ToolFileConverterBase>> BuiltInConverters = CreateBuiltInConverters();
+
+        public override ToolFileConverterBase CreateConverterCore(string toolFormat)
+        {
+            Lazy<ToolFileConverterBase> converter;
+            return BuiltInConverters.TryGetValue(toolFormat, out converter)
+                ? converter.Value
+                : null;
+        }
+
+        private static Dictionary<string, Lazy<ToolFileConverterBase>> CreateBuiltInConverters()
+        {
+            var result = new Dictionary<string, Lazy<ToolFileConverterBase>>();
+            // TODO: Reflect to get them all.
```

Comment:
As we discussed: added unit tests for this.

---
In reply to: [120986166](https://github.com/Microsoft/sarif-sdk/pull/582#discussion_r120986166) [](ancestors = 120986166)